ts-tests/country-coder.spec.ts

```diff
@@ -0,0 +1,1978 @@
+    });
+
+    it('codes Bir Tawil', () => {
+      expect(coder.wikidataQID('Bir Tawil')).toBe('Q620634');
+    });
+  });
+
+  // this doesn't need extensive tests since it's just a fetcher using `feature`
+  describe('emojiFlag', () => {
+    it('codes location in officially-assigned country: New York, United States as 🇺🇸', () => {
+      expect(coder.emojiFlag([-74, 40.6], { level: 'country' })).toBe('🇺🇸');
+    });
+
+    it('codes location in user-assigned, de facto country: Kosovo as 🇽🇰', () => {
+      expect(coder.emojiFlag([21, 42.6], { level: 'country' })).toBe('🇽🇰');
+    });
+
+    it('does not find for M49 code with no corresponding ISO code', () => {
+      expect(coder.emojiFlag('150')).toBeNull();
+    });
+
+    it('does not code North Pole', () => {
+      expect(coder.emojiFlag([0, 90], { level: 'country' })).toBeNull();
+    });
+  });
+
+  // this doesn't need extensive tests since it's just a fetcher using `feature`
+  describe('ccTLD', () => {
+    it('returns ccTLD in officially-assigned country: New York, United States as .us', () => {
+      expect(coder.ccTLD([-74, 40.6], { level: 'country' })).toBe('.us');
+    });
+
+    it('returns ccTLD in officially-assigned country: London, United Kingdom as .uk', () => {
+      expect(coder.ccTLD([0, 51.5], { level: 'country' })).toBe('.uk'); // not .gb
+    });
+
+    it('does not return a ccTLD for a region with no ccTLD', () => {
+      expect(coder.ccTLD('Bir Tawil')).toBeNull();
+    });
+
+    it('does not return a ccTLD for uncovered location North Pole', () => {
+      expect(coder.ccTLD([0, 90])).toBeNull();
+    });
+  });
+
+  describe('iso1A2Codes', () => {
+    it('codes locations', () => {
+      expect(coder.iso1A2Codes([-4.5, 54.2])).toStrictEqual(['IM', 'GB', 'UN']);
+      expect(coder.iso1A2Codes([-2.35, 49.43])).toStrictEqual(['GG', 'GB', 'UN']);
+      expect(coder.iso1A2Codes([-12.3, -37.1])).toStrictEqual(['TA', 'SH', 'GB', 'UN']);
+      expect(coder.iso1A2Codes([12.59, 55.68])).toStrictEqual(['DK', 'EU', 'UN']);
+      expect(coder.iso1A2Codes([2.35, 48.85])).toStrictEqual(['FX', 'FR', 'EU', 'UN']);
+      expect(coder.iso1A2Codes([-74, 40.6])).toStrictEqual(['US', 'UN']);
+      expect(coder.iso1A2Codes([21, 42.6])).toStrictEqual(['XK']);
+      expect(coder.iso1A2Codes([0, -90])).toStrictEqual(['AQ']);
+    });
+
+    it('codes bounding boxes', () => {
+      expect(coder.iso1A2Codes([-4.5, 54.2, -4.4, 54.3])).toStrictEqual(['IM', 'GB', 'UN']);
+      // area of US overlapping Canada's bounding box but not its polygon
+      expect(coder.iso1A2Codes([-74, 40.6, -71.3, 44.7])).toStrictEqual(['US', 'UN']);
+      // area overlapping both US and Canada
+      expect(coder.iso1A2Codes([-74, 40.6, -71.3, 45])).toStrictEqual(['CA', 'UN', 'US']);
+    });
+
+    it('does not code invalid arguments', () => {
+        // TODO: Evaluate how to supply non-type-safe API calls to TS members.
+        // expect(coder.iso1A2Codes([])).toStrictEqual([]);
```

Comment:
Here's one of two test lines that I removed/commented from the original .js-based tests. This is because those lines were no longer valid in Typescript-land. The iso1A2Codes() method expects different parameters than are supplied here, and so the test would not compile.

Reply:
I found this:
https://stackoverflow.com/questions/56705792/test-invalid-function-arguments-with-typescript-and-jestjs

It looks like the recommendation there is to add an `as any` to the argument to the test to get it to compile.

Reply:
That seems to work indeed! Excellent. I figured there had to be an elegant way to allow this...